docs/DeveloperGuide.md

```diff
@@ -1,34 +1,383 @@
 # Developer Guide
 
+- [Design & implementation](#design--implementation)
```

Comment:
Maybe you can consider to add numberings to the sections so that audience can follow better?